### PR TITLE
update bash completion to work more uniformly for linux/mac

### LINKED ISSUE #141: Rethink shell completion for macOS
The standard for most Linux distros seems to be `/etc/bash_completion.d/` but that location for completion files doesn't work for macOS. We should investigate setting some sane standards for macOS (and maybe others) and possibly do some os detection when generating them.

### PR DESCRIPTION
**Pull request by edaniszewski**
@timfallmk 

I think this should fix #141 - let me know what you think

### CODE REVIEW

File: pkg/commands/completion.go [hunk 1]
```diff
@@ -2,11 +2,46 @@ package commands
...
+complete -F _cli_bash_autocomplete synse
+`
+
+// zshCompletion is the zsh completion function for Synse. This is taken
+// from: https://github.com/urfave/cli/blob/master/autocomplete/zsh_autocomplete
+const zshCompletion = `
+_cli_zsh_autocomplete() {
+
+  local -a opts
+  opts=("${(@f)$(_CLI_ZSH_AUTOCOMPLETE_HACK=1 ${words[@]:0:#words[@]-1} --generate-bash-completion)}")
```
Comment (timfallmk) [inline pkg/commands/completion.go]: Nice env var naming.

Comment (edaniszewski) [reply pkg/commands/completion.go]: haha. wasn't me! I just copy and pasted what they had

**Comment by edaniszewski** [comment]
@tim - yeah, that's what I did here. I'm not sure its the "right" way to do it, but it seems cleaner to do it a single way that should work for both platforms rather than multiple different ways? maybe I'm wrong, but thats what PR reviews are for

**Comment by timfallmk** [comment]
Yeah, I'm not sure just replacing one for the other makes it any better. Furthermore, macOS can use several different methods (depending on what the user/other apps have set, or what's left around from previous installs), but the default is `.bash_profile` to my understanding. Additionally it seems kinda nasty to copy the _contents_ of the source generation script into their startup script (whatever that is), since it muddies it up. It's generally bad practice.

Here's my recommendation. We store the completion generation file _some place_ and source it in from the user's shell profile startup script. This is the way I've seen it done in the past. It also lets us detect and set some sane defaults for the most common platforms (e.g. `.bashrc`, `.bashprofile`, `/etc/bash_completion.d/` etc.).

For example, `gcloud` places the following in my `.bash_profile`:

```shell
# The next line updates PATH for the Google Cloud SDK.
if [ -f '/Users/timfall/Code/google-cloud-sdk/path.bash.inc' ]; then source '/Users/timfall/Code/google-cloud-sdk/path.bash.inc'; fi

# The next line enables shell command completion for gcloud.
if [ -f '/Users/timfall/Code/google-cloud-sdk/completion.bash.inc' ]; then source '/Users/timfall/Code/google-cloud-sdk/completion.bash.inc'; fi
```

The question then becomes where to store it. We've been just a binary so far, and I'm not sure I want to get away from that.

**tl;dr** Shell completion is kinda a legacy mess.

**Comment by timfallmk** [comment]
Now that I've written all that, I'm wondering if it's possible to have the binary itself spit out the completion script (maybe with a hidden flag). That would keep it simple and much more portable.

**Comment by timfallmk** [comment]
Per @edaniszewski 's scolding, just adding some checking logic for `.bash_profile` and `/etc/bash_completion.d` (with the latter being where we can just throw the script itself) based setups should be awesome.
